docs/4.2/gcp_guide.md

```diff
@@ -0,0 +1,203 @@
+
+### Computer Engine: Health Checks
+GCP relies heavily on [Health Checks](https://cloud.google.com/load-balancing/docs/health-checks), this is helpful when adding new instances to an 
+instance group. 
+
+To enable health checks in Teleport start with `teleport start --diag-addr=127.0.0.1:3000` see  [Admin Guide: Troubleshooting](admin-guide.md#troubleshooting) for more information. 
+
+### Storage: Cloud Firestore 
+
+[Cloud Firestore](https://cloud.google.com/firestore/) This storage backend uses Real-time updates to keep individual auth instances in sync and requires Firestore configured in Native mode.
+
+Add this storage configuration in teleport section of the config file (by default it's `/etc/teleport.yaml`):
+
+```
+teleport:
+  storage:
+    type: firestore
+    collection_name: cluster-data
+    project_id: gcp-proj-with-firestore-enabled
+    credentials_path: /var/lib/teleport/firestore_creds
+```
+
+### Storage: Google Cloud Storage
+
+
+When creating the Bucket we would recommend setting it up as `Dual-region` and with 
+`Standard` storage class. Provide access using a `Uniform` access control with a Google-managed
+key. 
+
+When setting up `audit_session_uri` use `gs://` session prefix.
+
+```
+storage:
+    ...
+    audit_sessions_uri: 'gs://teleport-session-storage-2?credentialsPath=/var/lib/teleport/gcs_creds&projectID=bc-jdurbin'
+    ...
+```
+
+### Network Services: Load Balancing
+
+Load Balancing is required for Proxy and SSH traffic `TCP Load Balancing` as Teleport requires
+custom ports for SSH and Web Traffic.
+
+### Network Services: Cloud DNS
+
+Cloud DNS is used to setup the public URL of the Teleport Proxy. Once setup an `A` 
+record is sufficient. 
+
+### Access: Service accounts 
+
+The Authentication server will need to read and write to Firestore for this it'll need to have
+the correct permission via Server Accounts. Learn how to [enable and create service accounts for instances](https://cloud.google.com/compute/docs/access/create-enable-service-accounts-for-instances).
+
+
+## GCP Quickstart
+
+## 1. Create Resources
+We recommend starting by creating the resources. We highly recommend creating these
+an infrastructure automation tool such as [Cloud Deployment Manager](https://cloud.google.com/deployment-manager/) or Terraform. 
+
+ 
+## 2. Install & Configure Teleport
+
+Follow install instructions from our [installation page](installation.md#linux). 
+ 
+We recommend configuring the Teleport in the below steps. 
+
+**1. Configure Teleport Auth Server** using the below example `teleport.yaml`, and start it using SystemD
+
+```yaml
+#
+# Sample Teleport configuration teleport.ymal file for Auth Server
+#
+teleport:
+  nodename: C02WG09CHTDH
+  data_dir: /var/lib/teleport
+  pid_file: /var/run/teleport.pid
+  auth_token: cluster-join-token
```

Comment:
This will do nothing on the auth server, you should set:

```
teleport:
  auth_service:
    tokens:
    - "proxy,node:EXAMPLE-CLUSTER-JOIN-TOKEN"
```